### ISSUE TITLE
Testing pure lambda functions locally

### ISSUE THREAD

**Issue by povils** [aws/chalice#736]
HI, I'm not sure how you guys are testing pure lambda functions locally? I think Chalice could integrate nicely with SAM local or something.

**Comment by dstufft** [comment]
I don't believe there is anything built into chalice for handling this use case now, so I'm going to mark this as a feature request for providing some way to test pure lambda functions locally, whether that is via integration with something else, or something native to chalice (or possibly just documenting how to assemble existing pieces to do this).

**Comment by wollerman** [comment]
It looks like #900 is the deployed lambda version of this request. Not sure if there was anything related to the local version though.

**Comment by Rubyj** [comment]
Any update on this?

**Comment by TraMZzz** [comment]
Any update ?

**Comment by nservidio** [comment]
Hey, would love this feature. I use a separate `@app.lambda_function()` to seed my database used by my `@app.route()`s. It's annoying to test at this point.

**Comment by beanaroo** [comment]
I would like to test async lambda invocations locally too. I have separate functions triggered by an incoming webhook request handler and being able to run that locally would be awesome.

**Comment by candypoplatte** [comment]
Any update on this?

**Comment by aimlessadam** [comment]
please consider this another request for this functionality

**Comment by jamesls** [comment]
Would people prefer integration with SAM local/docker (which gives a more accurate representation of the lambda runtime) or just some way to call the python lambda function locally that's "close enough"?  The tradeoff would be accuracy vs. ease of setup.

**Comment by Rubyj** [comment]
> Would people prefer integration with SAM local/docker (which gives a more accurate representation of the lambda runtime) or just some way to call the python lambda function locally that's "close enough"? The tradeoff would be accuracy vs. ease of setup.

I think either is acceptable.

**Comment by seerickcode** [comment]
Having the 'close enough' / 'just get my curl to the function" would be handy to have first.

**Comment by jamesls** [comment]
Ok, let's start with the "close-enough" approach initially.
